webknossos/tests/test_cli.py

```diff
@@ -0,0 +1,477 @@
+
+    prev_cwd = os.getcwd()
+    with TemporaryDirectory() as new_cwd:
+        os.chdir(new_cwd)
+        try:
+            yield
+        finally:
+            os.chdir(prev_cwd)
+
+
+MINIO_ROOT_USER = "TtnuieannGt2rGuie2t8Tt7urarg5nauedRndrur"
+MINIO_ROOT_PASSWORD = "ANTN35UAENTS5UIAEATD"
+MINIO_PORT = "8000"
+
+TESTDATA_DIR = Path(__file__).parent.parent / "testdata"
+
+
+@pytest.fixture(scope="module", name="remote_testoutput_path")
+def fixture_remote_testoutput_path() -> Iterator[UPath]:
+    """Minio is an S3 clone and is used as local test server"""
+    container_name = "minio"
+    cmd = (
+        "docker run"
+        f" -p {MINIO_PORT}:9000"
+        f" -e MINIO_ROOT_USER={MINIO_ROOT_USER}"
+        f" -e MINIO_ROOT_PASSWORD={MINIO_ROOT_PASSWORD}"
+        f" --name {container_name}"
+        " --rm"
+        " -d"
+        " minio/minio server /data"
+    )
+    subprocess.check_output(shlex.split(cmd))
+    remote_path = UPath(
+        "s3://testoutput",
+        key=MINIO_ROOT_USER,
+        secret=MINIO_ROOT_PASSWORD,
+        client_kwargs={"endpoint_url": f"http://localhost:{MINIO_PORT}"},
+    )
+    remote_path.fs.mkdirs("testoutput", exist_ok=True)
+    try:
+        yield remote_path
+    finally:
+        subprocess.check_output(["docker", "stop", container_name])
+
+
+@pytest.mark.skipif(
+    sys.platform != "linux",
+    reason="Only run this test on Linux, because it requires a running `minio` docker container.",
+)
+def test_tiff_cubing_zarr_s3(remote_testoutput_path: UPath) -> None:
+    """Tests zarr support when performing tiff cubing."""
+    out_path = remote_testoutput_path / "tiff_cubing"
+
+    result = runner.invoke(
+        app,
+        [
+            "convert",
+            "--jobs",
+            "2",
+            "--voxel-size",
+            "11.24,11.24,25",
+            "--data-format",
+            "zarr",
+            str(TESTDATA_DIR / "tiff"),
+            str(out_path),
+        ],
+        env={
+            "AWS_SECRET_ACCESS_KEY": MINIO_ROOT_PASSWORD,
+            "AWS_ACCESS_KEY_ID": MINIO_ROOT_USER,
+            "S3_ENDPOINT_URL": f"http://localhost:{MINIO_PORT}",
+        },
+    )
+
+    assert result.exit_code == 0
+
+    assert (out_path / "tiff").exists()
+    assert (out_path / "tiff" / "1").exists()
+
+    assert (out_path / "tiff" / "1" / ".zarray").exists()
+    assert (out_path / PROPERTIES_FILE_NAME).exists()
+
+    with (out_path / PROPERTIES_FILE_NAME).open("r") as file, (
+        TESTDATA_DIR / "tiff" / "datasource-properties.zarr-fixture.json"
+    ).open("r") as fixture:
+        json_a = json.load(file)
+        json_fixture = json.load(fixture)
+        del json_a["id"]
+        del json_fixture["id"]
+        assert json_a == json_fixture
+
+
+def test_main() -> None:
+    """Tests the functionality of the webknossos command."""
+
+    result_without_args = runner.invoke(app, [])
+    assert result_without_args.exit_code == 0
+
+
+def test_check_equality() -> None:
```

Comment:
Can you add a test to verify that 2 datasets that differ are detected? Please make a new dataset that is an exact copy except for 1 value somewhere.

Reply:
Test has been added.